chain/client/src/client.rs

```diff
@@ -102,7 +104,35 @@ pub struct Client {
     /// again to prevent network from stalling if a large percentage of the network missed a block
     last_time_head_progress_made: Instant,
     /// Keeps track of when the latest blocks and chunks were received.
-    chunks_delay_tracker: ChunksDelayTracker,
+    pub chunks_delay_tracker: ChunksDelayTracker,
+}
+
+// Debug information about the upcoming block.
+#[derive(Default)]
+pub struct UpcomingBlockDebugStatus {
+    // How long is this block 'in progress' (time since we first saw it).
+    pub in_progress_for: Option<Duration>,
+    // How long is this block in orphan pool.
+    pub in_orphan_for: Option<Duration>,
+    // Epoch id for the block.
+    pub epoch_id: EpochId,
+    // List of chunk hashes that belong to this block.
+    pub chunk_hashes: Vec<ChunkHash>,
+
+    // Chunk statuses are below:
+    // We first sent the request to fetch the chunk
+    // Later we get the response from the peer and we try to reconstruct it.
+    // If reconstructions suceeds, the chunk will be marked as complete.
+    // If it fails (or fragments are missing) - we're going to re-request the chunk again.
+
+    // Chunks that we reqeusted (sent the request to peers).
+    pub chunks_requested: HashSet<ChunkHash>,
+    // Chunks for which we've received the response.
+    pub chunks_received: HashSet<ChunkHash>,
```

Comment:
Not sure what's the difference between "chunk_received" and "chunk_not_completed" and "chunks_completed". It seems from the comments that "chunks_received" means some chunk parts are received, so it could be overlapping with "chunks_not_completed" and "chunks_completed". Please update comments to make it clearer

Reply:
you're right - we can remove the 'not completed' HashSet.

Reply:
Actually I think it's good to add another state "chunks_missing" if there is no chunk included for this shard, so we don't have to wait for it

Reply:
If chunk is not in the block - then it will be missing from chunk_hashes Vec - so it should be then skipped during print.

We could actually do something else: and in such case actually print a different symbol - would be useful if the RPC node doesn't track all the shards - but I'd do that in a separate PR.

Reply:
Actually not in your current implementation. When chunks are missing, the block still contains chunk headers for all shards, for the shards that don't have new chunks, the chunk header will simply be the chunk header in the previous block. So to check when a chunk is missing or not, you need to check if "chunk.height_included() == block.height()"

Reply:
I see - in such case, the current code will mark such chunks as 'succeeded' (as it will see that they already exist in storage).

We might add a separate char in the future - if this one turns out to be too confusing.